pallets/utxo/src/lib.rs

```diff
@@ -595,6 +754,25 @@ pub mod pallet {
                     ensure!(!<UtxoStore<T>>::contains_key(hash), "output already exists");
                     new_utxos.push(hash.as_fixed_bytes().to_vec());
                 }
+                Destination::StakeExtra(_) => {
+                    ensure!(output.value > 0, "output value must be nonzero");
+                    let hash = tx.outpoint(output_index);
+                    new_utxos.push(hash.as_fixed_bytes().to_vec());
+                    ensure!(!<LockedUtxos<T>>::contains_key(hash), "output already exists");
+                }
+                Destination::Stake {..} => {
+                    ensure!(output.value >= T::MinimumStake::get(), "output value must be equal or more than the set minimum stake");
+                    let hash = tx.outpoint(output_index);
+                    ensure!(!<LockedUtxos<T>>::contains_key(hash), "output already exists");
+                    new_utxos.push(hash.as_fixed_bytes().to_vec());
+                }
+                Destination::WithdrawStake{ outpoints, pub_key } => {
+                    ensure!(output.value == outpoints.len() as Value , "output value must be equivalent to the number of utxos to withdraw.");
```

Comment:
Can you elaborate this check here? Isn't `output.value` some amount MLT we've staked and that we want to withdraw now? Either that's not true or I don't understand why we're comparing that value to the number of `outpoints`? Why are these two different types of numbers being compared?

Reply:
Although now I'm thinking:
@muursh  do we allow withdrawal of a portion of the stake, or should it always be **all** of it?

Because if it's only a portion, meaning i have to change how the **unlock_stake** works.

Reply:
For now let's only support an entire withdrawal, they can re-stake a smaller amount if they decide they want to